app/main.py

```diff
@@ -2,24 +2,118 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None :
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: object,
+            name: str
+    ) -> None :
+        self.protected_name = f"_{name}"
+
+    def validate_amount(self, value: int) -> None:
+        if not type(value) == int:
+            raise TypeError("Value must be an integer type")
+
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"{value} is out of range"
+                             f" [{self.min_amount}...{self.max_amount}]")
+
+    def __set__(
+            self,
+            instance: object,
+            value: int
+    ) -> None:
+        self.validate_amount(value)
+        setattr(instance, self.protected_name, value)
+
+    def __get__(self, instance: object, owner: object) -> int :
+        return getattr(instance, self.protected_name)
 
 
 class Visitor:
-   pass
+    def __init__(
+        self,
+        name: str,
+        age: int,
+        weight: int,
+        height: int
+    ) -> None :
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
+
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        super().__init__(age, weight, height)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
+
+    def __init__(
+            self,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        super().__init__(age, weight, height)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: type
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool :
+        try:
+            if self.limitation_class == ChildrenSlideLimitationValidator:
```

Comment:
Instead of directly comparing `self.limitation_class` with `ChildrenSlideLimitationValidator`, consider using `issubclass` to check if `self.limitation_class` is a subclass of `SlideLimitationValidator`. This will make the code more flexible and maintainable.